### PR TITLE
feature: add support for mining presets

### PR DESCRIPTION
**Pull request by b-rowan**
Adds support for mining mode presets (used by vnish and luxos mainly), as well as a flag, `supports_presets` to check against.  Presets will be included in the config, including available presets, so that they can be selected from and sent to the miner.

Hopefully a partial prefix to #245.

### CODE REVIEW

File: pyasic/config/mining/presets.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+    def as_vnish(self) -> dict:
+        if self.name is not None:
+            return {"preset": self.name}
+        return {}
+
+    @classmethod
+    def from_vnish(cls, web_preset: dict):
+        name = web_preset["name"]
+        hr_power_split = web_preset["pretty"].split("~")
```
Comment (wilfredallyn) [inline pyasic/config/mining/presets.py:19]: The first dict in `web_presets` has `Disabled` for the `pretty` field so it doesn't have a 2nd element after the split and causes error when you access `hr_power_split[1]`

```python
[{'name': 'disabled',
  'pretty': 'Disabled',
  'status': 'untuned',
  'modded_psu_required': False},
 {'name': '1740',
  'pretty': '1740 watt ~ 65 TH',
  'status': 'tuned',
  'modded_psu_required': False},
 ...
```

This is the error
```python
File ~/Documents/pyasic/pyasic/config/mining/presets.py:21, in MiningPreset.from_vnish(cls, web_preset)
     19 hr_power_split = web_preset["pretty"].split("~")
     20 power = hr_power_split[0].replace("watt", "").strip()
---> 21 hashrate = hr_power_split[1].replace("TH", "").strip()
     22 tuned = web_preset["status"] == "tuned"
     23 modded_psu = web_preset["modded_psu_required"]

IndexError: list index out of range
```

This fixes it
```python
def from_vnish(cls, web_preset: dict):
    name = web_preset["name"]
    hr_power_split = web_preset["pretty"].split("~")
    
    if len(hr_power_split) == 1:
        return cls(
            name=name,
            power=None,
            hashrate=None,
            tuned=False,
            modded_psu=False,
        )
        
    # rest of code same ...
```

Comment (b-rowan) [reply pyasic/config/mining/presets.py:19]: I think we can expect `Disabled` will be a separate mode represented in pyasic, so maybe I just skip that altogether?

**Comment by wilfredallyn** [comment]
I pulled the recent commits, ran the code, and was able to parse the presets. The issue with the `Disabled` preset is fixed, and everything else looks good to me.

**Review by wilfredallyn — APPROVED** [review]
I pulled the recent commits, ran the code, and was able to parse the presets. The issue with the `Disabled` preset is fixed, and everything else looks good to me.

**Pull request merged**